desc/objectives/_fast_ion.py

```diff
@@ -359,3 +360,263 @@ def _compute_spline(self, params, constants=None):
+    )
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "max-J: "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        w0=1,
+        w1=4,
+        thresh0=-1,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        jac_chunk_size=None,
+        name="maxJ",
+        grid=None,
+        X=16,
+        Y=32,
+        # Y_B is expensive to increase if one does not fix num well per transit.
+        Y_B=None,
+        alpha=np.array([0.0]),
+        num_transit=20,
+        num_well=None,
+        num_quad=32,
+        num_pitch=64,
+        pitch_batch_size=None,
+        surf_batch_size=1,
+    ):
+        if target is None and bounds is None:
+            target = 0.0
+
+        self._grid = grid
+
+        self._X = X
+        self._Y = Y
+        Y_B = setdefault(Y_B, 2 * Y)
+        self._thresh0 = thresh0
+        self._w0 = w0
+        self._w1 = w1
+
+        self._constants = {
+            "quad_weights": 1.0,
+            "alpha": alpha,
+            "thresh0": self._thresh0,
+            "w0": self._w0,
+            "w1": self._w1,
+        }
+
+        self._hyperparam = {
+            "Y_B": Y_B,
+            "num_transit": num_transit,
+            "num_well": setdefault(num_well, Y_B * num_transit),
+            "num_quad": num_quad,
+            "num_pitch": num_pitch,
+            "pitch_batch_size": pitch_batch_size,
+            "surf_batch_size": surf_batch_size,
+        }
+        self._key = "dJ_ds"
+
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+            jac_chunk_size=jac_chunk_size,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+        if self._grid is None:
+            self._grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, sym=False)
+        assert self._grid.can_fft2
+        self._constants["clebsch"] = FourierChebyshevSeries.nodes(
+            self._X,
+            self._Y,
+            self._grid.compress(self._grid.nodes[:, 0]),
+            domain=(0, 2 * np.pi),
+        )
+        self._constants["fieldline quad"] = leggauss(self._hyperparam["Y_B"] // 2)
+        self._constants["quad"] = get_quadrature(
+            leggauss(self._hyperparam.pop("num_quad")),
+            (automorphism_sin, grad_automorphism_sin),
+        )
+        self._dim_f = self._grid.num_rho
```

Comment:
Make sure that this matches the size of the array returned in `objective.compute`

Reply:
It does. The test_objective_compute test ensures that.

Reply:
No it does not. For example this will fail if I give a grid with multiple surfaces since the objective compute returns a single value.